### PR TITLE
docker-compose: add possibility for a local Nginx module.

### PR DESCRIPTION
**Pull request by teresanovielloatinstana**
# WHAT
- Modify docker-compose file to use a local Nginx module
- Modify docker-compose file to enable the debug of agent's spans

# WHY
Improvements

### CODE REVIEW

**Review by sriemer — CHANGES_REQUESTED** [review]
Thanks for your work on this one, Teresa. :smiley: 
Please address the required changes that I've noticed. TIA

Also have you noticed the high duplication in your short git commit messages?
"Modify docker-compose file to " should be shrunk to "docker-compose: ".

* `docker-compose: enable the debug of agent's spans`
* `docker-compose: use a local Nginx module`

This is much better and faster readable.

**Comment by sriemer** [comment]
You are actually only adding the possibility for a local NGINX module.
So the commit should be `docker-compose: add possibility for a local Nginx module`.
I renamed the PR accordingly.

File: docker-compose.yml [hunk 1]
```diff
@@ -47,9 +50,13 @@ services:
     volumes:
       - /var/run:/var/run
       - /run:/run
-      - /dev:/dev:ro
-      - /sys:/sys:ro
-      - /var/log:/var/log:ro
+      - /dev:/dev
+      - /sys:/sys
+      - /var/log:/var/log
```
[the remark this answers is not in the document]
Comment (teresanovielloatinstana) [reply docker-compose.yml:55]: Thanks Sebastian,
it was just distraction from my side to leave these volumes mounted.
I will change it and test again.

File: docker-compose.yml [hunk 2]
```diff
@@ -24,6 +24,9 @@ services:
     volumes:
       - ./nginx/nginx.conf:/etc/nginx/nginx.conf
       - ./nginx/instana-config.json:/etc/instana-config.json
+      # Include the following lines to override the nginx module, for local testing
+      #- ./nginx/ngx_http_opentracing_module.so:/opt/instana/nginx/ngx_http_opentracing_module.so
```
[the remark this answers is not in the document]
Comment (teresanovielloatinstana) [reply docker-compose.yml:28]: Thanks Sebastian, will do

**Comment by teresanovielloatinstana** [comment]
> docker-compose: add possibility for a local Nginx module

Thanks Sebastian, I have produced the requested changes and solved the relative conversations. Also, I have modified the .gitignore file.
I have tested again on this branch and the test runs smoothly.
I will squash with the commit message `docker-compose: add possibility for a local Nginx module`

File: docker-compose.yml [hunk 3]
```diff
@@ -24,6 +24,9 @@ services:
     volumes:
       - ./nginx/nginx.conf:/etc/nginx/nginx.conf
       - ./nginx/instana-config.json:/etc/instana-config.json
+      # Include the following lines to override the nginx module, for local testing
```
Comment (sriemer) [inline docker-compose.yml:27]:
```suggestion
      # Instana developers only: Include the following lines to override the nginx module, for local testing
```

File: README.md [hunk 4]
```diff
@@ -19,6 +19,8 @@ agent_zone=<name of the zone for the agent; default: nginx-tracing-demo>
 
 ## Build & Launch
 
+If you wish to test an NGINX tracing module built locally on your host, you can copy it into `nginx/ngx_http_opentracing_module.so` and uncomment the volumne mounting in the `nginx` service in the `docker-compose.yml` file.
+
```
Comment (sriemer) [inline README.md:23]: This whole repo is public and customer facing. This stuff is closed source. Customers cannot build their own modules. They can only receive some from us.

I'd like to drop this whole paragraph here. Let's keep it undocumented for not confusing customers.

```suggestion
```

File: README.md [hunk 5]
```diff
@@ -30,7 +32,7 @@ This will build and launch
 - `nginx` service, which routes all incoming requests to the ...
 - `server-app` service, a simple Spring Boot application that returns `200` to any HTTP request.
 
-After the agent is bootstrapped and starts accepting spans from NGINX, the resulting traces in the Analyze view will look like the following:
+After the agent is bootstrapped and starts accepting spans from NGINX, the resulting traces are logged into files named `spans-<timestamp_ns>` into the directory `agent/logs` and in the Analyze view will look like the following:
```
Comment (sriemer) [inline README.md:35]: Too complex sentence. Needs to be split.
```suggestion
After the agent is bootstrapped and starts accepting spans from NGINX, the resulting traces are logged into files named `spans-<timestamp_ns>` in the directory `agent/logs`.

In the Analyze view this looks like the following:
```

**Review by sriemer — CHANGES_REQUESTED** [review]
Thanks Teresa, just some finishing touches.
Please notify me once the issues I've found are addressed. TIA

File: README.md (hunk 4 above)
Comment (teresanovielloatinstana) [reply README.md]: Removed, thanks

File: README.md (hunk 5 above)
Comment (teresanovielloatinstana) [reply README.md]: Thanks Sebastian it is much better, I will try my best to keep paragraphs simple.

**Review by sriemer — APPROVED** [review]
LGTM now. Thanks.

**Pull request merged**